dbms/src/Server/Server.cpp

```diff
@@ -1091,17 +1082,31 @@ int Server::main(const std::vector<std::string> & /*args*/)
     global_context->getSharedContextDisagg()->use_autoscaler = use_autoscaler;
 
     /// Init File Provider
-    bool enable_encryption = false;
+    // ps_write is a UniversalPageStoragePtr, now it is nullptr, pass it by reference to init KeyspacesKeymanager.
```

Comment:
The `ps_write` is used to initialize `KeyspacesKeymanager` now, and inited later. However, in `KeyspacesKeymanager` I found is it not guarded by any checks of `nullptr`. Maybe it is guarded by some implicit design that no call will be invoked to this `KeyspacesKeymanager`. But since it is implicit, it could be broken by someone else.
I think we have to add checks in `KeyspacesKeymanager`.
Meanwhile, a better way is to manage to make `KeyspacesKeymanager` inited with a always valid `ps_write`.

Reply:
it is a bug here, will fix later...

Reply:
fixed